index.js

```diff
@@ -20,8 +20,60 @@ mongoose
     return self.connection.dropDatabase();
   })
   .then(() => {
-    // Run your code here, after you have insured that the connection was made
+    let firstIteration = Recipe.create(data[0])
+     firstIteration
+        .then((res)=>{
+              console.log("data was sent",res)
+    })
+        .catch(()=>{
+              console.log("failure")
+    })
+  
+  
+  let secondIteration = Recipe.insertMany(data)
+    secondIteration
+        .then((res)=>{
+              res.filter((res)=>{ console.log(res.title)})
+      
+               console.log("data was sent")
+    })
+        .catch(()=>{
+              console.log("failure")
+      
+    })
+  
+
+
+  Promise.all([firstIteration,secondIteration])
+
+  .then(()=>{
+    let thirdIteration =Recipe.findOneAndUpdate({title:"Rigatoni alla Genovese"},{$set:{duration:100}})
+  thirdIteration
+          .then((res)=>{
+                console.log("data was sent",res)
+               })
+          .catch(()=>{
+                console.log("failure")
+               })
+    let fourthIteration =  Recipe.deleteOne({title:'Carrot Cake'})
+               fourthIteration
+                    .then((res)=>{
+                            console.log("data was sent",res)
+                         })
+                   .catch(()=>{
+                            console.log("failure")
+                         })
+                       })
+
+         Promise.all([thirdIteration, fourthIteration])
+          .then(() => {
+            mongoose.connection.close()
+            . then(() => {
+
```

Comment:
don't leave yout .then() empty. if you don't put anything in there then you don' need them in the first place. These .then and .catch are not needed for the promise to be executed. They only exist to run some code after the promise was either resolved or rejected.